website/site/content/docs/custom-previews.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Custom Previews
+position: 50
+---
+
+# Customizing the Preview Pane
+
+The preview pane displays an HTML preview of the content you're editing, and can be customized to
+display content identical to your production site using templates and CSS.
+
+## Preview Templates
+
+Preview templates are regular templates that are updated in realtime as an entry is being
+edited. Netlify CMS uses [template compilers](#) to support different template languages. React
+components are supported without any compiler, and a Handlebars compiler is included by
+default.
+
+### Writing a Basic Template
+
+Let's look at an example template written in Handlebars. To create a template for use with a
+collection, we'll start by observing the `fields` configuration for the collection:
+
+```yaml
+# this snippet is from the main Netlify CMS configuration file, typically config.yml
+
+fields:
+  - { name: title, label: Title, widget: string }
+  - { name: text, label: Text, widget: text }
+```
+
+A preview template displaying these fields might look like this:
+
+```handlebars
+<div>
+  <h1>{{ title }}</h1>
+  <p>{{ text }}</p>
+</div>
+```
+
+You can add styles and classes as necessary to make the preview match your production site.
+
+
+### Registering Templates
+
+Once you have a template, you'll need to register it to be used with a collection. We're
+using a handlebars template, so we'll only need to provide a name for the template and the template
+itself.
+
+```js
+// You can pull in your template in a number of ways, but we'll go with an
+// inline template for simplicity.
+const template = `
+<div>
+  <h1>{{ title }}</h1>
+  <p>{{ text }}</p>
+</div>
+`
+
+CMS.registerPreviewTemplate('blog', template)
+```
+
+Registering a template may require other arguments depending on your template compiler. For more
+information on `registerPreviewTemplate`, check out the [API docs](#), as well as the docs for your
```

Comment:
Will we allow the preview template to be pulled in via a filename?

```js
CMS.registerPreviewTemplate('blog', 'template.hbs');
```

If so, do we need to allow an option/switch for that so that we can easily differentiate between filenames and templates?

Reply:
I wasn't planning on it - the case where someone can actually use production templates in the CMS is pretty rare. It's most likely to be possible for projects using React components as templates, but in that case the file is going to be ES6/JSX, so the CMS still can't use it as is. We'd also have to grab templates via GitHub since they wouldn't be available in the deployed site.

Reply:
Although this could also be used buy sites that do have custom templates for the CMS, but no build system for easily importing them in a registration script. This is worth considering.